subprojects/docs/src/docs/userguide/build_cache.adoc

```diff
@@ -225,6 +229,24 @@ include::sample[dir="buildCache/cacheable-bundle/groovy",files="build.gradle[tag
 include::sample[dir="buildCache/cacheable-bundle/kotlin",files="build.gradle.kts[tags=bundle-task]"]
 ====
 
+=== Using annotations
+
+Instead of using the runtime API, it is actually better to create a custom task class which is cacheable using <<more_about_tasks.adoc#sec:task_input_output_annotations,annotations to declare the inputs and outputs>>.
+
+We continue the example from above, trying to get a cacheable bundle task.
+Given that we already have the `NpmTask`, we now create a subclass declaring the inputs and outputs.
```

Comment:
Should we mention the case where you already have properties declared as inputs, but they are not relocatable because of a lack of `@PathSensitivity` and how you can _override_ the base property as `@Internal` and declare a new one, maybe `protected` tagged with the necessary annotation?